### PR TITLE
Upgrade wagtail2.x

### PR DESCRIPTION
**Pull request by mojeto**
work in progress

### CODE REVIEW

File: core/templates/core/includes/menu.html [hunk 1]
```diff
@@ -9,14 +9,14 @@
...
-    <li class='nav__item'>
-        <a href="{% url 'wagtailsearch_search' %}" class="nav__link{% if request.path == '/search/' %} is-active{% endif %}">
-            <span class="u-hide--small">Search</span>
-            {% if not request.path == '/search/' %}
-                {% set color = "teal" %}
-            {% endif %}
-            {% include "core/includes/icon.html" with icon="search" class="i--small u-middle i--hover" %}
-        </a>
-    </li>
+{#    <li class='nav__item'>#}
```
Comment (haydngreatnews) [inline core/templates/core/includes/menu.html:12]: Note to self to check on this

File: core/wagtail_hooks.py [hunk 2]
```diff
@@ -1,51 +1,67 @@
...
     return css_includes
 
 
-@hooks.register('register_rich_text_features')
+@hooks.register("register_rich_text_features")
 def register_cleanhtml_feature(features):
     features.register_editor_plugin(
-        'hallo', 'cleanhtml',
+        "hallo",
+        "cleanhtml",
```
Comment (haydngreatnews) [inline core/wagtail_hooks.py:31]: We'll need to stick with Hallo (not migrate to `draft`) if this is an important thing

Comment (haydngreatnews) [reply core/wagtail_hooks.py:31]: but I'm tempted to say it's not at all important, and may be in the core Draft functionality if my hunch is correct about it being for cleaning stuff pasted from Word

**Review by haydngreatnews — COMMENTED** [review]
I'll work through the things I've raised here

**Comment by haydngreatnews** [comment]
Bugs:
- [ ] Display of Springload SVG is weird
- [ ] Display of Search icon SVG is weird

File: core/wagtail_hooks.py (hunk 2 above)
Comment (haydngreatnews) [reply core/wagtail_hooks.py:31]: Confirmed, I will remove this configuration, as word-tidying is something Draft takes care of

File: core/templates/core/includes/menu.html (hunk 1 above)
Comment (haydngreatnews) [reply core/templates/core/includes/menu.html:12]: Resolved -- Search view added, and plumbed in correctly

File: core/models.py [hunk 3]
```diff
@@ -175,19 +193,19 @@ class WagtailPage(Page):
...
     feed_image = models.ForeignKey(
-        'wagtailimages.Image',
+        "wagtailimages.Image",
         null=True,
         blank=True,
         on_delete=models.SET_NULL,
-        related_name='+'
+        related_name="+",
     )
-    body = RichTextField(blank=True, features=['bold', 'italic', 'ol', 'ul', 'link', 'cleanhtml'])
```
Comment (AndrewCalderSpringload) [inline core/models.py:190]: Curious, why was **cleanhtml** removed?

Comment (haydngreatnews) [reply core/models.py:190]: We've removed the feature altogether, as it's Hallo.js-specific. Wagtail 2 moves to Draft.js, and that does all the same things but by default.

All `cleanhtml` really does is clean up the garbage you get when you paste stuff from Word.

https://github.com/springload/madewithwagtail/pull/170/commits/24c702ce7b09d81de5cb6da1d6ea75627a66f5e3 has all the changes for that bundled together

File: docker-compose.yml [hunk 4]
```diff
@@ -16,7 +16,7 @@ services:
         environment:
             VIRTUAL_HOST: "madewithwagtail.dev.springload.nz"
             VIRTUAL_PORT: 8000
-            VIRTUAL_MEDIA_DOMAIN: "madewithwagtail.org/media"
+            VIRTUAL_MEDIA_DOMAIN: "https://d1nvwtjgmbo5v5.cloudfront.net/media/"
```
Comment (AndrewCalderSpringload) [inline docker-compose.yml:19]: Is d1nvwtjgmbo5v5.cloudfront.net likely to change?

**Review by AndrewCalderSpringload — APPROVED** [review]

File: docker-compose.yml (hunk 4 above)
Comment (haydngreatnews) [reply docker-compose.yml:19]: Nah, Cloudfront ones only change when the distribution is recreated, and that's a very deliberate infra-y thing to do

**Pull request merged**